specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01/redis.json

```diff
@@ -2697,6 +3007,167 @@
+          "readOnly": true,
+          "enum": [
+            "Updating",
+            "Succeeded",
+            "Deleting",
+            "Deleted",
+            "Canceled",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "AccessPolicyProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "type": {
+          "type": "string",
+          "description": "Built-In or Custom access policy",
+          "readOnly": true,
+          "enum": [
+            "Custom",
+            "BuiltIn"
+          ],
+          "x-ms-enum": {
+            "name": "AccessPolicyType",
+            "modelAsString": true
+          }
+        },
+        "permissions": {
+          "type": "string",
+          "description": "Permissions for the access policy. Learn how to configure permissions at https://aka.ms/redis/aad/accesspolicypermissions"
+        }
+      },
+      "required": [
+        "permissions"
+      ],
+      "description": "All properties of an access policy.",
+      "type": "object"
+    },
+    "RedisCacheAccessPolicyRequestResponse": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RedisCacheAccessPolicyProperties",
+          "description": "Properties of an access policy."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "Response to get/put access policy.",
+      "type": "object"
+    },
+    "RedisCacheAccessPolicyList": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RedisCacheAccessPolicyRequestResponse"
+          },
+          "description": "List of access policies (with properties) of a Redis cache."
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Link for next set."
+        }
+      },
+      "description": "List of access policies (with properties) of a Redis cache.",
+      "type": "object"
+    },
+    "RedisCacheAccessPolicyAssignment": {
+      "description": "Properties for an access policy assignment",
+      "properties": {
+        "objectId": {
+          "type": "string",
+          "description": "Object Id to assign access policy to"
+        },
+        "objectIdAlias": {
+          "type": "string",
+          "description": "User friendly name for object id. Also represents username for token based authentication"
+        }
+      },
+      "required": [
+        "objectId",
+        "objectIdAlias"
+      ],
+      "type": "object"
+    },
+    "RedisCacheAccessPolicyAssignmentProperties": {
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "description": "Provisioning state of an access policy assignment",
```

Comment:
Now these are really properties of the set (or group) of access policy assignments, we should tweak the naming and/or description e.g.

```
    "RedisCacheAccessPolicyAssignmentSetProperties": {
      "properties": {
        "provisioningState": {
          "type": "string",
          "description": "Provisioning state of an access policy assignment set",
```

Reply:
Updated naming and description. Additionally removed RequestResponse from AccessPolicy